### ISSUE TITLE
Terraform remote state push -force fails serial check with Consul backend

### ISSUE THREAD

**Issue by OttensPavel** [hashicorp/terraform#26662]
### Terraform Version
```
0.12.29
```

### Terraform Configuration Files
```terraform
terraform {
  backend "consul" {
    address = "localhost:8500"
    scheme  = "http"
    path    = "personal/tf_state_push_with_force_bug"
  }
  
  required_providers {
    azurerm = {
	  version         = "=2.21.0"
	  subscription_id = "<Please don't forget put here your Azure Subscription Id>"
    }
  }
}

provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "rg" {
  name     = "myTFResourceGroup"
  location = "westus2"
}

resource "azurerm_resource_group" "rg2" {
  name     = "myTFResourceGroup2"
  location = "westus2"
}
```

### Expected Behavior
The state should be pushed to the remote and forced over any existing state

### Actual Behavior
```bash
$ terraform state push -force terraform.tfstate.backup
Failed to write state: cannot import state with serial 0 over newer state with serial 1
```

### Steps to Reproduce
```bash
#!/bin/bash
set -o errexit

# Init and apply
terraform init
terraform apply -auto-approve

# Backup state
terraform state pull > terraform.tfstate.backup

# Remove one resource
terraform state rm azurerm_resource_group.rg2

# Check serial counters
echo "Serial of current state:"
terraform state pull | grep "serial"

echo "Serial of state backup:"
cat terraform.tfstate.backup | grep "serial"

# Restore state from backup
echo "Try to restore state from backup ..."
terraform state push -force terraform.tfstate.backup
```

### Additional Context
Without Consul backend this bug doesn't reproduce.
Also I've not reproduce this bug on Terraform v.0.13.*.

**Environment:**
Windows 10, WSL1, Ubuntu 18.04
Consul Docker image: consul:1.8.4
```bash
docker run -it --rm -p 8500:8500 -p 8502:8502 consul:1.8.4
```

**Comment by alisdair** [comment]
Hi @OttensPavel. When you say you've not reproduced the bug on 0.13, does that mean that you haven't tried, or that it appears to be fixed?

We don't intend to release any more 0.12 versions, so if the bug is fixed in 0.13 then we'd encourage you to upgrade when you're able to.

**Comment by OttensPavel** [comment]
Hi @alisdair.
I've tried, - this bug doesn't reproduce in 0.13.4.

At this time we don't have time to upgrade, but we have plans to do so.

**Comment by alisdair** [comment]
I'm afraid that the answer then is that the solution is to go through the upgrade process to 0.13 to get the fix for the backend. Thanks again for reporting the issue, and I'm sorry I can't offer an easier way forward!

**Comment by remilapeyre** [comment]
Hi, I looked a bit into this bug, it is a duplicate of https://github.com/hashicorp/terraform/issues/25761 which has been fixed by Pam Selle in https://github.com/hashicorp/terraform/pull/26190. The fix was backported to 0.13 in https://github.com/hashicorp/terraform/pull/26201 but not to 0.12 so upgrading to 0.13 is indeed the best way forward :)
